src/main/java/com/gluonhq/substrate/util/ios/InfoPlist.java

```diff
@@ -0,0 +1,347 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.substrate.util.ios;
+
+import com.dd.plist.NSArray;
+import com.dd.plist.NSDictionary;
+import com.dd.plist.NSString;
+import com.dd.plist.PropertyListParser;
+import com.gluonhq.substrate.Constants;
+import com.gluonhq.substrate.model.ProcessPaths;
+import com.gluonhq.substrate.model.ProjectConfiguration;
+import com.gluonhq.substrate.util.FileOps;
+import com.gluonhq.substrate.util.Logger;
+import com.gluonhq.substrate.util.ProcessRunner;
+import com.gluonhq.substrate.util.XcodeUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+
+public class InfoPlist {
+
+    private static final List<String> assets = new ArrayList<>(Arrays.asList(
+            "Default-375w-667h@2x~iphone.png", "Default-414w-736h@3x~iphone.png", "Default-portrait@2x~ipad.png",
+            "Default-375w-812h-landscape@3x~iphone.png", "Default-568h@2x~iphone.png", "Default-portrait~ipad.png",
+            "Default-375w-812h@3x~iphone.png", "Default-landscape@2x~ipad.png", "Default@2x~iphone.png",
+            "Default-414w-736h-landscape@3x~iphone.png", "Default-414w-896h@3x~iphone.png", "Default-414w-896h-landscape@3x~iphone.png",
+            "Default-landscape~ipad.png", "iTunesArtwork",
+            "iTunesArtwork@2x"
+    ));
+
+    private static final List<String> iconAssets = new ArrayList<>(Arrays.asList(
+            "Contents.json", "Gluon-app-store-icon-1024@1x.png", "Gluon-ipad-app-icon-76@1x.png", "Gluon-ipad-app-icon-76@2x.png",
+            "Gluon-ipad-notifications-icon-20@1x.png", "Gluon-ipad-notifications-icon-20@2x.png",
+            "Gluon-ipad-pro-app-icon-83.5@2x.png", "Gluon-ipad-settings-icon-29@1x.png", "Gluon-ipad-settings-icon-29@2x.png",
+            "Gluon-ipad-spotlight-icon-40@1x.png","Gluon-ipad-spotlight-icon-40@2x.png","Gluon-iphone-app-icon-60@2x.png",
+            "Gluon-iphone-app-icon-60@3x.png","Gluon-iphone-notification-icon-20@2x.png", "Gluon-iphone-notification-icon-20@3x.png",
+            "Gluon-iphone-spotlight-icon-40@2x.png", "Gluon-iphone-spotlight-icon-40@3x.png", "Gluon-iphone-spotlight-settings-icon-29@2x.png",
+            "Gluon-iphone-spotlight-settings-icon-29@3x.png"
+    ));
+
+    private final XcodeUtils.SDKS sdk;
+    private final ProjectConfiguration projectConfiguration;
+    private final ProcessPaths paths;
+    private final XcodeUtils xcodeUtil;
+
+    private final Path appPath;
+    private final Path rootPath;
+    private final Path tmpPath;
+
+    private Path partialPListDir;
+    private String bundleId;
+    private String minOSVersion = "11.0";
+
+    public InfoPlist(ProcessPaths paths, ProjectConfiguration projectConfiguration, XcodeUtils.SDKS sdk) throws IOException {
```

Comment:
Validate arguments

Reply:
done, like this?
```
if (paths == null || projectConfiguration == null) {
            throw new IllegalArgumentException("Error, argument can't be null");
        }
        ```

Reply:
I would use `Objects.requireNonNull`?

Reply:
Yes, I though about that, but that throws an NPE?